README.md

```diff
@@ -1 +1,81 @@
 # Immutable Web Apps
+
+## Introduction
+
+_Immutable Web Applications_ is a framework-agnostic methodology for building and deploying static [single-page application](https://en.wikipedia.org/wiki/Single-page_application) that:
+
+- Minimizes risk and complexity of live releases.
+- Simplifies and maximizes caching.
+- Minimizes the need for servers and systems administration.
+- Enables continuous delivery through simple, flexible, atomic deployments.
+
+## Principles
+
+The methodology is based on the principles of ___strictly separating___:
+
+- Configuration from code.
+- Release tasks from build tasks.
+- Dynamic content from static content.
+
+These principles are inspired by [The Twelve-Factor App](https://12factor.net/).
+
+## Concepts
+
+The following concepts define the core requirements for an _Immutable Web Application_. They are framework and infrastructure agnostic.
+
+### _Permabundles_
+
+Permabundles are the static assets that make up a single-page application.
+
+- They must be available at a [permalink](https://en.wikipedia.org/wiki/Permalink) that is uniquely versioned or fingerprinted and _independent of the web application environment(s)_.
+
+- They must be configured for long-term caching. (`cache-control: public, max-age=31536000, immutable`)
+
+- __They must not contain any environment-specific configuration.__
+
+### `index.html` <span style="font-size:larger;">___is___</span> configuration
+
+ An `index.html` exists for each deployment environment.
+
+- They must contain fully-qualified references to the _permabundle_ assets.
+
+- They must never be cached by the browser or a public cache that cannot be purged on-demand. (`cache-control: no-store`)
```

Comment:
Though i mostly agree with this, I could definitely question whether or not this is a core principle. I think some form of known TTL is important here, but you could make the argument that having a 1 day TTL on index.html is feasible when combined with backwards compatible APIs and the permabundles not going away. People would just gradually transition over from the old version to the new one over the course of a day.

Reply:
While I agree that there are some interesting use cases where you could have a 1 day TTL on `index.html`, it does eliminate the possibility of doing an atomic rollback which I do think is a core principle.